visidata/loaders/mysqlclient.py

```diff
@@ -0,0 +1,49 @@
+"""Loader to use visidata in mysql (or mycli) client. Use the following config to enable
+visidata by default in the mysql client:
+
+Edit ~/.my.cnf:
+[mysql]
+silent
+pager=vd -f mysqlclient
+"""
+
+from visidata import SequenceSheet
+import io
+
+
+def load_mysqlclient(fn):
+    vs = open_mysqlclient(Path(fn))
+    yield from vs.iterload()
+
+def open_mysqlclient(p):
+    return MysqlClientSheet(p.name, source=p)
+
+def _mysqlclient_split_lines(fp):
+    buf = ''
+    while True:
+        nextbuf = fp.readline()
+        if not nextbuf:
+            break
+        buf += nextbuf
+
+        *rows, buf = buf.split("\n")
+        yield from rows
+
+    yield from buf.split("\n")
+
+
+# rowdef: list
+class MysqlClientSheet(SequenceSheet):
+    def iterload(self):
+        with io.TextIOWrapper(self.source.open("rb"), errors="replace", newline="\n", encoding=self.options.getonly('encoding', 'global', 'utf-8')) as fp:
```

Comment:
can we remove this and just do `with self.source.open_text() as fp:` ?

Reply:
nope, we need the newline="\n" option
also the open_text method of Path is pretty evil. it contains:
```
        if self.fptext:
            self.rfile = RepeatFile(self.fptext)
            return self.rfile
```

so if the newline="\n" option is ignored, as fptext already exists.
i think we should avoid passing fptext to the Path constructor as much as possible, and always try to pass a binary stream. the binary can be easily converted to text, the other way around not.